File: ene-jun-2021/Jose Adrian Ontiveros Moran/Ordinario/Ordi2.py
def binarySearch(lista, izquierda, derecha, objetivo):
    if derecha >= izquierda:
        medio = izquierda+(derecha-izquierda)//2
        if int(lista[medio][1]) == objetivo:
            return medio
        elif int(lista[medio][1]) > objetivo:
            return binarySearch(lista,izquierda,medio-1,objetivo)
        else:
            return binarySearch(lista,medio+1,derecha,objetivo)
    else:
        return -1
entrada = input('ingresa numero y plumon a buscar').split(' ')
n = int(entrada[0])
s = entrada[1]
lista = []
for i in range(n):
    entrada_plumon = input('').split(' ')
    lista.append(entrada_plumon)

    if s == entrada_plumon[0]:
        valor_numerico = int(entrada_plumon[1])

posicion = binarySearch(lista, 0, len(lista)-1, valor_numerico)+1
if posicion == -1:
    print('no esta')
else:
    print("numero es")
    print(posicion) 

File: ago-dic-2022/SofiaGC/ordi2.java
/*En este problema lo que se busca es encontrar el articulo que Panchito quiere
 * lo que se puede lograr a traves de la busqueda binaria ya que la lista de 
 * articulos ya esta ordenada. Lo que hara la busqueda binaria es partir la lista en 2
 * y buscara primero el articulo en la mitad, si no ira a la mitad derecha o a la izquierda 
 * dependiendo que tan grande sea el articulo dentro de la lista
 */

import java.util.Scanner;
public class ordi2 {
    public static void main(String[] args) {
        Scanner teclado = new Scanner(System.in);

        System.out.println("Escribe el numero de articulos que hay en la lista ");
        int numArticulos = teclado.nextInt();
        System.out.println();
        
        teclado.nextLine();

        System.out.println(" Escribe el nombre del articulo que busca Panchito ");
        String panchito = teclado.nextLine();
        System.out.println();

        System.out.println("Escribe la lista de los articulos junto con su precio ");
        String listaArticulos[] = new String[numArticulos];
        String entrada = teclado.nextLine();
        for (int i = 0; i < numArticulos; i++) {
            listaArticulos[i] = teclado.nextLine();
        } 
        System.out.println();

        int a = binarySearch(listaArticulos, panchito);
        System.out.println("El articulo se encuentra en la posicion " + a + " de la lista de articulos");
    }
    static int binarySearch(String[] arr, String x){
        int l = 0, r = arr.length - 1;
        while (l <= r) {
            int m = l + (r - l) / 2;
            int res = x.compareTo(arr[m]);
 
            // Revisa si el articulo esta en la mitad
            if (res == 0)
                return m;
 
            // Si el articulo es mayor se ignora la mitad izquierda
            if (res > 0)
                l = m + 1;
 
            // Si el articulo es menor se ignora la mitad derecha
            else
                r = m - 1;
        }
 
        return -1;
    }
}
